knative-operator/pkg/controller/knativeserving/knativeserving_controller.go

```diff
@@ -212,3 +223,20 @@ func finalizerName() string {
 	}
 	return name
 }
+
+func (r *ReconcileKnativeServing) updateStatus(desired *servingv1alpha1.KnativeServing) error {
+	ks := &servingv1alpha1.KnativeServing{}
+	err := r.client.Get(context.TODO(), types.NamespacedName{Name: desired.Name, Namespace: desired.Namespace}, ks)
+	if err != nil {
+		return err
+	}
+
+	// If there's nothing to update, just return.
+	if equality.Semantic.DeepEqual(ks.Status, desired.Status) {
```

Comment:
There's something about always doing this `DeepEqual` after always doing the same `DeepEqual` (different object) in the caller that tickles my spidey sense.

Reply:
Yeah, the second one is probably not needed here.